docassemble/ALToolbox/save_input_data.py

```diff
@@ -0,0 +1,52 @@
+from docassemble.webapp.jsonstore import (
+    read_answer_json,
+    write_answer_json,
+    JsonStorage,
+    JsonDb,
+)
+from docassemble.base.generate_key import random_alphanumeric
+from docassemble.base.functions import get_current_info
+from docassemble.base.util import variables_snapshot_connection
+import random
+
+__all__ = ["save_input_data"]
+
+
+def save_input_data(title="", input_dict=None, tags=None):
+    """
+    This function is used by survey type interviews to save input data for data reporting purposes.
+    input_dict must be structured like this:
+    my_dict['fld_name'] = [fld_name, fld_type as string], e.g.
+    input_fields_dict['event_rating'] = [event_rating, 'radio']
+    """
+    type_dict = dict()
+    field_dict = dict()
+    for k, v in input_dict.items():
+        type_dict[k] = list(v)[1]
+        field_dict[k] = list(v)[0]  # Field name without quotes
+
+    data_to_save = dict()
+    data_to_save["title"] = title
+    data_to_save["field_type_list"] = type_dict  # Needed later in process_data
+
+    for k, v in type_dict.items():
+        # If a field is of checkboxes type, flatten its elements dict
+        # so that each key/value pair is saved in its own column.
+        if v in ["checkboxes", "multiselect"]:
```

Comment:
Suggestion: we could eliminate the extra dictionary (storing the types) if we use a check like this: `isinstance(item, DADict)`

Reply:
Hmm. What about `data_to_save['field_type_list'] = type_dict` which is part of the data to be saved, and later used in the Stats? Or maybe I misunderstood what you meant?

Reply:
@purplesky2016 check out https://github.com/SuffolkLITLab/docassemble-ALToolbox/pull/68/commits/4714fb07d18640f3d574979e3805ead4d289b592 when you have a chance--this is the same output without making the author specify the type